frontend/src/metabase/reference/metrics/MetricDetailContainer.jsx

```diff
@@ -37,19 +32,39 @@ const mapDispatchToProps = {
 )
 export default class MetricDetailContainer extends Component {
   static propTypes = {
+    router: PropTypes.shape({
+      replace: PropTypes.func.isRequired,
+    }).isRequired,
     params: PropTypes.object.isRequired,
     location: PropTypes.object.isRequired,
     user: PropTypes.object.isRequired,
     metric: PropTypes.object.isRequired,
     metricId: PropTypes.number.isRequired,
     databaseId: PropTypes.number.isRequired,
-    isEditing: PropTypes.bool,
   };
 
+  constructor(props) {
+    super(props);
+    this.startEditing = this.startEditing.bind(this);
+    this.endEditing = this.endEditing.bind(this);
+  }
+
   async fetchContainerData() {
     await actions.wrappedFetchMetricDetail(this.props, this.props.metricId);
   }
 
+  startEditing() {
+    const { metric, router } = this.props;
+    router.replace(`/reference/metrics/${metric.id}/edit`);
```

Comment:
`replace` matches the previous behavior, but `push` might be better here. I kinda like being able to go back after clicking "Edit". 

@mazameli or @kdoh do we have a general philosophy on what's a new page? Should switching to edit push a new history item?